### ISSUE TITLE
Using `export as` can cause false positives

### ISSUE THREAD

**Issue by RobertMic** [art0rz/eslint-plugin-no-barrel-files#2]
First off, thank you for writing this rule, it's very helpful!

I believe I noticed a bug where this rule can incorrectly report an error if you use `export { ... as ...}`. I don't think this should be counted as a re-export:

```ts
// a.tsx
export function Foo() { return <>Foo</> }

// b.tsx
import {Foo} from './a';
import {memo} from 'react';

function FooWrapper() { return <><Foo/></>;}

const memoized = memo(FooWrapper);
// This line is marked as a re-export, but shouldn't be
export {memoized as Foo};
```

**Comment by art0rz** [comment]
Hi, thanks for your contribution!

You are absolutely right. I'll look into it.

**Comment by art0rz** [comment]
fix included in new release, let me know if it works correctly. added this as a test case as well

**Comment by RobertMic** [comment]
Works as expected now, thank you so much : )
